kernel/tests/row_tracking.rs

```diff
@@ -0,0 +1,599 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use itertools::Itertools;
+use object_store::{path::Path, ObjectStore};
+use serde_json::{Deserializer, Value};
+use tempfile::{tempdir, TempDir};
+use url::Url;
+
+use delta_kernel::arrow::array::{Int32Array, Int64Array, StringArray};
+use delta_kernel::arrow::record_batch::RecordBatch;
+use delta_kernel::engine::arrow_conversion::TryIntoArrow;
+use delta_kernel::engine::arrow_data::ArrowEngineData;
+use delta_kernel::engine::default::executor::tokio::TokioBackgroundExecutor;
+use delta_kernel::engine::default::DefaultEngine;
+use delta_kernel::schema::{DataType, SchemaRef, StructField, StructType};
+use delta_kernel::transaction::CommitResult;
+use delta_kernel::{DeltaResult, Error, Snapshot};
+
+use test_utils::{create_table, engine_store_setup, test_read};
+
+/// Helper function to create a simple table with row tracking enabled.
+async fn create_row_tracking_table(
```

Comment:
Additional tests:
- Check that we assign base row ids / default row commit version even when the table property `delta.enableRowTracking` is disabled or missing (as long as the rowTracking table feature is present on the table)
- Removing files, then adding new files
- Empty commit & committing an empty file - not sure if that's allowed, which is in itself a good reason to cover it
- No commit: the table shouldn't have a domainMetadata action in that case

Reply:
I added additional tests for the cases that (1) a transaction contains no add files and (2) contains empty add files.

The other scenarios are future work IMO since kernel does not support removing files and creating tables (i.e., no commits) at the moment.